lib/ansible/modules/remote_management/manageiq/manageiq_alert_profiles.py

```diff
@@ -0,0 +1,319 @@
+  name:
+    description:
+      - The unique alert profile name in ManageIQ.
+      - Required when state is "absent" or "present".
+  description:
+    description:
+      - User-visible profile name (will be shown in the ManageIQ UI).
+      - Will be identical to "name" if not specified.
+  resource_type:
+    description:
+      - The resource type for the alert profile in ManageIQ. Required when state is "present".
+    choices: ['Vm', 'ContainerNode', 'MiqServer', 'Host', 'Storage', 'EmsCluster',
+              'ExtManagementSystem', 'MiddlewareServer']
+  alerts:
+    description:
+      - List of alert descriptions to assign to this profile.
+      - Required if state is "present"
+  notes:
+    description:
+      - Optional notes for this profile
+
+'''
+
+EXAMPLES = '''
+- name: List alert profiles in ManageIQ
+  manageiq_alert_profiles:
+    state: list
+    manageiq_connection:
+      url: 'http://127.0.0.1:3000'
+      username: 'admin'
+      password: 'smartvm'
+      verify_ssl: False
+
+- name: Add an alert profile to ManageIQ
+  manageiq_alert_profiles:
+    state: present
+    name: Test profile
+    resource_type: ContainerNode
+    alerts:
+      - Test Alert 01
+      - Test Alert 02
+    manageiq_connection:
+      url: 'http://127.0.0.1:3000'
+      username: 'admin'
+      password: 'smartvm'
+      verify_ssl: False
+
+- name: Delete an alert profile from ManageIQ
+  manageiq_alert_profiles:
+    state: absent
+    name: Test profile
+    manageiq_connection:
+      url: 'http://127.0.0.1:3000'
+      username: 'admin'
+      password: 'smartvm'
+      verify_ssl: False
+'''
+
+RETURN = '''
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.manageiq import ManageIQ, manageiq_argument_spec
+
+
+class ManageIQAlertProfiles(object):
+    """ Object to execute alert profile management operations in manageiq.
+    """
+
+    def __init__(self, manageiq):
+        self.manageiq = manageiq
+
+        self.module = self.manageiq.module
+        self.api_url = self.manageiq.api_url
+        self.client = self.manageiq.client
+        self.url = '{api_url}/alert_definition_profiles'.format(api_url=self.api_url)
+
+    def get_profiles(self):
+        """ Get all alert profiles from ManageIQ
+        """
+        try:
+            response = self.client.get(self.url + '?expand=alert_definitions,resources')
+        except Exception as e:
+            self.module.fail_json(msg="Failed to query alert profiles: {error}".format(error=e))
+        return response.get('resources') or []
+
+    def get_alerts(self, alert_descriptions):
+        """ Get a list of alert hrefs from a list of alert descriptions
+        """
+        alerts = []
+        for alert_description in alert_descriptions:
+            alert = self.manageiq.find_collection_resource_or_fail("alert_definitions",
+                                                                   description=alert_description)
```

Comment:
I think this should now match by `name` (and method arg should be named `alert_names`)

Reply:
Ignore this, I confused alert profile — matched by name, because that's the unique field — with alerts that are matched by description, which is unique in that table and more user friendly.